larpandoradlcontent/LArDeepLearning/DeepLearningTrackShowerIdAlgorithm.h

```diff
@@ -0,0 +1,50 @@
+/**
+ *  @file   larpandoradlcontent/LArMonitoring/DeepLearningTrackShowerIdAlgorithm.h
+ *
+ *  @brief  Header file for the deep learning track shower id algorithm.
+ *
+ *  $Log: $
+ */
+#ifndef LAR_DEEP_LEARNING_TRACK_SHOWER_ID_ALGORITHM_H
+#define LAR_DEEP_LEARNING_TRACK_SHOWER_ID_ALGORITHM_H 1
+
+#include "Pandora/Algorithm.h"
+
+namespace lar_content
+{
+
+/**
+ *  @brief  DeepLearningTrackShowerIdAlgorithm class
+ */
+class DeepLearningTrackShowerIdAlgorithm: public pandora::Algorithm
+{
+public:
+    /**
+     *  @brief  Default constructor
+     */
+    DeepLearningTrackShowerIdAlgorithm();
+    virtual ~DeepLearningTrackShowerIdAlgorithm();
+
+private:
+    typedef std::map<const pandora::CaloHit*, std::tuple<int, int, int, int>> CaloHitToPixelMap;
+    typedef std::map<int, int> PixelToTileMap;
+
+    pandora::StatusCode Run();
+    pandora::StatusCode Train();
+    pandora::StatusCode Infer();
+    pandora::StatusCode ReadSettings(const pandora::TiXmlHandle xmlHandle);
+    void GetHitRegion(const pandora::CaloHitList& caloHitList, float& xMin, float& xMax, float& zMin, float& zMax);
+    void GetSparseTileMap(const pandora::CaloHitList& caloHitList, const float xMin, const float zMin, const float tileSize,
+        const int nTilesX, PixelToTileMap& sparseMap);
```

Comment:
Typically place `&` by variable, rather than type, but don't enforce this as rigorously in content libraries as in SDK. Could revisit the .clang-format file to enforce this easily. Whilst on this subject, don't forget to check for whitespace, using removeWhitespace.sh too.